### PR TITLE
Enabling the endpoint slices if the cluster supports

### PR DESCRIPTION
**Pull request by jayaramsatya**
acc-provision should provision enable_endpointslices flag as true
enpoint slice Obj is retrived for default Kubernetes service
to verify cluster supports Endpoint slices.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -331,14 +333,21 @@ func (cont *AciController) Init() {
...
-	if cont.config.EnabledEndpointSlice {
+	endpointSlice := false
+	kubeClient := cont.env.(*K8sEnvironment).kubeClient
+	esobj, err := kubeClient.DiscoveryV1beta1().EndpointSlices("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+	if err == nil && esobj != nil {
 		cont.serviceEndPoints = &serviceEndpointSlice{}
 		cont.serviceEndPoints.(*serviceEndpointSlice).cont = cont
-	} else {
+		endpointSlice = true
+		cont.log.Info("Service endpointslice Interface is initialized")
```
Comment (snaiksat) [inline pkg/controller/controller.go:345]: suggest rephrasing -
Initializing ServiceEndpointSlices

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -331,14 +333,21 @@ func (cont *AciController) Init() {
...
+	}
+	if !endpointSlice {
 		cont.serviceEndPoints = &serviceEndpoint{}
 		cont.serviceEndPoints.(*serviceEndpoint).cont = cont
+		cont.log.Info("Service enpoints Interface is initialized")
```
Comment (snaiksat) [inline pkg/controller/controller.go:350]: typo, also suggest rewording:
Initializing ServiceEndpoints

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -331,14 +333,21 @@ func (cont *AciController) Init() {
...
 	cont.log.Debug("Initializing IPAM")
 	cont.initIpam()
-	//@FIXME need to set this value based on feature capability supported by cluster
+	// check if the cluster supports endpoint slices
 	// if cluster doesn't have the support fallback to endpoints
-	if cont.config.EnabledEndpointSlice {
+	endpointSlice := false
+	kubeClient := cont.env.(*K8sEnvironment).kubeClient
+	esobj, err := kubeClient.DiscoveryV1beta1().EndpointSlices("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+	if err == nil && esobj != nil {
```
Comment (snaiksat) [inline pkg/controller/controller.go:341]: Can this logic be refactored so that it can be used commonly in the host agent as well (as opposed to duplicating the code)?

File: pkg/controller/controller.go (hunk 1 above)
Comment (jayaramsatya) [reply pkg/controller/controller.go:345]: done

File: pkg/controller/controller.go (hunk 3 above)
Comment (jayaramsatya) [reply pkg/controller/controller.go:341]: This is single line to get the endpoint slice Obj, I guess it should be ok, reset of the code is initialize the  kubeClient is specific to controller or host agent

File: pkg/controller/controller.go (hunk 2 above)
Comment (jayaramsatya) [reply pkg/controller/controller.go:350]: done

File: pkg/controller/controller.go (hunk 3 above)
Comment (jayaramsatya) [reply pkg/controller/controller.go:341]: let me move it to common this function

**Review by snaiksat — APPROVED** [review]
